backend/src/test/java/ai/verta/modeldb/GlobalSharingTest.java

```diff
@@ -0,0 +1,451 @@
+                                          ? ModelDBResourceEnum.ModelDBServiceResourceTypes
+                                              .REPOSITORY
+                                          : resourceType)
+                                  .build())
+                          .setService(ServiceEnum.Service.MODELDB_SERVICE)
+                          .build())
+                  .setVisibility(resourceVisibility)
+                  .setCollaboratorType(customPermission.getCollaboratorType())
+                  .setCanDeploy(customPermission.getCanDeploy())
+                  .build());
+        }
+        try {
+          switch (resourceType) {
+            case PROJECT:
+              client2ProjectServiceStub.getProjectById(
+                  GetProjectById.newBuilder().setId(orgResourceId).build());
+              break;
+            case DATASET:
+              client2DatasetServiceStub.getDatasetById(
+                  GetDatasetById.newBuilder().setId(orgResourceId).build());
+              break;
+            case REPOSITORY:
+            default:
+              repository =
+                  client2RepositoryServiceStub
+                      .getRepository(
+                          GetRepositoryRequest.newBuilder()
+                              .setId(
+                                  RepositoryIdentification.newBuilder()
+                                      .setRepoId(repository.getId()))
+                              .build())
+                      .getRepository();
+              break;
+          }
+          if (!isReadAllowed) {
+            fail();
+          }
+        } catch (StatusRuntimeException e) {
+          if (isReadAllowed) {
+            throw e;
+          }
+        }
+        try {
+          switch (resourceType) {
+            case PROJECT:
+              client2ProjectServiceStub.addProjectTag(
+                  AddProjectTag.newBuilder()
+                      .setId(orgResourceId)
+                      .setTag("new-tag" + new Date().getTime())
+                      .build());
+              break;
+            case DATASET:
+              client2DatasetServiceStub.updateDatasetName(
+                  UpdateDatasetName.newBuilder()
+                      .setId(orgResourceId)
+                      .setName("new_name" + new Date().getTime())
+                      .build());
+              break;
+            case REPOSITORY:
+            default:
+              client2RepositoryServiceStub.updateRepository(
+                  SetRepository.newBuilder()
+                      .setId(
+                          RepositoryIdentification.newBuilder()
+                              .setRepoId(Long.parseLong(orgResourceId)))
+                      .setRepository(
+                          repository
+                              .toBuilder()
+                              .setDescription("new_description" + new Date().getTime()))
+                      .build());
```

Comment:
Why are these 3 paths testing setting different things? I'm confused.

Reply:
No reason. Why? I have no preferred way to do it. I've updated the code to use only description.

Reply:
Thanks! I was wondering if there was any gain or issue with testing a single path.